src/shared/quota-util.c

```diff
@@ -40,3 +43,41 @@ int quotactl_path(int cmd, const char *path, int id, void *addr) {
 
         return quotactl_devnum(cmd, devno, id, addr);
 }
+
+int is_proj_id_quota_supported(int fd, uint32_t proj_id, struct dqblk *req) {
+        int r = 0;
+
+        r = RET_NERRNO(quotactl_fd(fd, QCMD_FIXED(Q_GETQUOTA, PRJQUOTA), proj_id, req));
+        if (r == -ESRCH || ERRNO_IS_NEG_NOT_SUPPORTED(r) || ERRNO_IS_NEG_PRIVILEGE(r))
+                return false;
+
+        if (r < 0)
+                return r;
+
+        return true;
+}
+
+int set_quota_proj_id(int fd, const char *path, uint32_t proj_id) {
```

Comment:
maybe rename this to set_proj_id_verify_exclusive() to make clear that this is mostly just a wrapper around set_proj_id() that howwever checks exclusivity